deepmd/infer/deep_pot.py

```diff
@@ -229,30 +233,29 @@ def eval(
             The atomic energy. Only returned when atomic == True
         atom_virial
             The atomic virial. Only returned when atomic == True
+        descriptor
+            Descriptors. Only returned when eval_descriptor == True
         """
         # reshape coords before getting shape
         natoms = len(atom_types)
         coords = np.reshape(np.array(coords), [-1, natoms * 3])
         numb_test = coords.shape[0]
-        if atomic:
-            if self.modifier_type is not None:
+        if self.auto_batch_size is not None:
+            def eval_func(*args, **kwargs):
+                return self.auto_batch_size.execute_all(self._eval_inner, numb_test, natoms, *args, **kwargs)
+        else:
+            eval_func = self._eval_inner
+        output = eval_func(coords, cells, atom_types, fparam = fparam, aparam = aparam, atomic = atomic, efield = efield, eval_descriptor=eval_descriptor)
+
+        if self.modifier_type is not None:
+            if atomic:
                 raise RuntimeError('modifier does not support atomic modification')
-            if self.auto_batch_size is not None:
-                return self.auto_batch_size.execute_all(self._eval_inner, numb_test, natoms,
-                           coords, cells, atom_types, fparam = fparam, aparam = aparam, atomic = atomic, efield = efield)
-            return self._eval_inner(coords, cells, atom_types, fparam = fparam, aparam = aparam, atomic = atomic, efield = efield)
-        else :
-            if self.auto_batch_size is not None:
-                e, f, v = self.auto_batch_size.execute_all(self._eval_inner, numb_test, natoms,
-                              coords, cells, atom_types, fparam = fparam, aparam = aparam, atomic = atomic, efield = efield)
-            else:
-                e, f, v = self._eval_inner(coords, cells, atom_types, fparam = fparam, aparam = aparam, atomic = atomic, efield = efield)
-            if self.modifier_type is not None:
-                me, mf, mv = self.dm.eval(coords, cells, atom_types)
-                e += me.reshape(e.shape)
-                f += mf.reshape(f.shape)
-                v += mv.reshape(v.shape)
-            return e, f, v
```

Comment:
Any reason why `auto_batch_size` is removed?

Reply:
Not removed. Instead, it's moved into a wrapper to make the code more clear.